service/frontend/workflow_handler.go

```diff
@@ -2631,6 +2637,27 @@ func (wh *WorkflowHandler) CreateSchedule(ctx context.Context, request *workflow
 	}, nil
 }
 
+func (wh *WorkflowHandler) validateScheduledWorkflowSearchAttributes(
+	request *workflowservice.CreateScheduleRequest,
+	namespaceName namespace.Name,
+) error {
+
+	scheduledAction := request.GetSchedule().GetAction().GetAction()
+	if scheduledAction == nil {
+		panic("no scheduled workflow")
+	}
+	action, ok := scheduledAction.(*schedpb.ScheduleAction_StartWorkflow)
+	if !ok {
+		panic(scheduledAction)
```

Comment:
there may be other action types in the future, you can't panic here

Reply:
We should switch on the type and return an error in the `default` case, no?